src/DotNet.Status.Web/TimelineIssueTriageService.cs

```diff
@@ -0,0 +1,286 @@
+
+                    // add into notification epic -> currently 8/2020 its First Response epic
+                    NotificationEpicOptions epic = _githubOptions.Value.NotificationEpic;
+
+                    if (epic != null)
+                    {
+                        var epicRepoData = await gitHubClient.Repository.Get(_githubOptions.Value.Organization, epic.Repository);
+
+                        _logger.LogInformation("Adding the issue to ZenHub Epic...");
+                        await _zenHub.AddIssueToEpicAsync(
+                            new ZenHubClient.IssueIdentifier(issuePayload.Repository.Id, issue.Number),
+                            new ZenHubClient.IssueIdentifier(epicRepoData.Id, epic.IssueNumber)
+                        );
+                    }
+                    else
+                    {
+                        _logger.LogInformation("No ZenHub epic configured, skipping...");
+                    }
+                }
+
+                updatedCategory = "InTriage";
+            }
+
+            if (issuePayload.Action == "closed")
+            {
+                IReadOnlyList<IssueComment> comments = gitHubClient.Issue.Comment.GetAllForIssue(issuePayload.Repository.Id, issuePayload.Issue.Number).Result;
+
+                foreach (var comment in comments)
+                {
+                    // Look for category information in comment
+                    string category = GetTriageIssueProperty("category", comment.Body);
+                    if (!string.IsNullOrEmpty(category))
+                    {
+                        updatedCategory = category;
+                    }
+                }
+            }
+
+            foreach (var triageItem in triageItems)
+            {
+                triageItem.Url = issuePayload.Issue.Html_Url;
+                if (updatedCategory != null)
+                {
+                    triageItem.UpdatedCategory = updatedCategory;
+                    triageItem.Url = issuePayload.Issue.Html_Url;
+                }
+                _logger.LogInformation($"buildId: {triageItem.BuildId}, recordId: {triageItem.RecordId}, index: {triageItem.Index}, category: {triageItem.UpdatedCategory}, url: {triageItem.Url}");
+            }
+
+            await IngestTriageItemsIntoKusto(triageItems);
+
+            await gitHubClient.Issue.Comment.Create(issuePayload.Repository.Id, issuePayload.Issue.Number, $"Bot has updated the 'TimelineIssuesTriage' database.\n**PowerBI reports may take up to 24 hours to refresh**\n\nSee {_docLink} for more information.");
+
+            return;
+        }
+
+        private async Task IngestTriageItemsIntoKusto(ICollection<TriageItem> triageItems)
+        {
+            _logger.LogInformation("Entering IngestTriageItemIntoKusto");
+            await KustoHelpers.WriteDataToKustoInMemoryAsync(
+                _kustoIngestClientFactory.GetClient(),
+                _kustoOptions.Value.Database,
+                "TimelineIssuesTriage",
```

Comment:
Please extract and name this const

Reply:
Majority of code has been copy/pasted from darcbot (now deleted) project. I am not sure if I shall, in such situation, refactor everything.

Reply:
@tkapin shall I?